src/messaging/helpers/email-envelope.js

```diff
@@ -0,0 +1,168 @@
+import Utils from '../../utils'
+
+function checkCriteria(criteria) {
+  if (criteria) {
+    throw new Error('Addresses can not be set if criteria is already set in a builder')
+  }
+}
+
+export default class EmailEnvelop {
```

Comment:
@stanislav-grin the word "Envelop" is misspelled here, there should be "e" on the end: "Envelope". Not sure if this will impact the API because everywhere else it does say "Envelope"

Reply:
Thanks. Fixed in next commit.
https://github.com/Backendless/JS-SDK/pull/130/commits/bde6d43c66a27d5ef8805bb35ad0a05e5ecf2d74